src/main/java/oss/fosslight/service/impl/T2UserServiceImpl.java

```diff
@@ -64,608 +60,623 @@
+  }
+
+  @Override
+  public int modifyUserRoles(T2Users t2Users) {
+    // 0. 사용자 ID가 없을 경우 0 리턴
+    if (null == t2Users.getUserId() || t2Users.getUserId().trim().equals("")) {
+      return 0;
+    }
+
+    // 1. 사용자의 모든 역할 삭제
+    authMapper.deleteAuthoritiesByUserId(t2Users);
+
+    // 2. 모든 사용자가 가지고 있는 기본 권한 추가
+    T2Roles defaultRole = new T2Roles();
+    defaultRole.setupDefaultRoleData();
+    T2Authorities authority = new T2Authorities();
+    authority.setAuthority(defaultRole.getAuthority());
+    authority.setUserId(t2Users.getUserId());
+
+    authMapper.insertAuthorities(authority);
+
+    // 3. 사용자의 역할 1건씩 등록
+    List<T2Authorities> userAuthorityList = t2Users.getAuthoritiesList();
+
+    if (null != userAuthorityList) {
+      for (T2Authorities authorityItem : userAuthorityList) {
+        authorityItem.setUserId(t2Users.getUserId());
+        authMapper.insertAuthorities(authorityItem);
+      }
+    }
+
+    return 1;
+  }
+
+  @SuppressWarnings("serial")
+  @Override
+  public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+    return new UserDetails() {
+      @Override
+      public boolean isEnabled() {
+        return false;
+      }
+
+      @Override
+      public boolean isCredentialsNonExpired() {
+        return false;
+      }
+
+      @Override
+      public boolean isAccountNonLocked() {
+        return false;
+      }
+
+      @Override
+      public boolean isAccountNonExpired() {
+        return false;
+      }
+
+      @Override
+      public String getUsername() {
+        return null;
+      }
+
+      @Override
+      public String getPassword() {
+        return null;
+      }
+
+      @Override
+      public Collection<? extends GrantedAuthority> getAuthorities() {
+        return null;
+      }
+    };
+  }
+
+  @Override
+  public String getPassword(T2Users user) {
+    T2Users result = userMapper.getPassword(user);
+
+    if (result != null) {
+      return result.getPassword();
+    }
+
+    return null;
+  }
+
+  /**
+   * 유저 목록 조회
+   */
+  @Override
+  public Map<String, Object> getUserList(T2Users t2Users) {
+    int records = userMapper.selectUserTotalCount(t2Users);
+    t2Users.setCurPage(t2Users.getPage());
+    t2Users.setPageListSize(t2Users.getRows());
+    t2Users.setTotListSize(records);
+    List<T2Users> userList = userMapper.selectParamUserList(t2Users);
+
+    HashMap<String, Object> map = new HashMap<String, Object>();
+    map.put("page", t2Users.getPage());
+    map.put("total", t2Users.getTotBlockSize());
+    map.put("records", records);
+    map.put("rows", userList);
+
+    return map;
+  }
+
+  /**
+   * 유저 저장
+   */
+  @Override
+  @Transactional
+  public void setUser(T2Users vo) throws Exception {
+    vo.setCreator(vo.getUserId());
+    vo.setModifier(vo.getUserId());
+
+    userMapper.insertUsers(vo);
+    userMapper.insertAuthority(vo);
+  }
+
+  @Override
+  public void modUser(List<T2Users> vo) {
+    for (T2Users t2Users : vo) {

```

Comment:
I replaced **for-loop** with **enhanced for-loop** to improve clarity and readability.